### PR TITLE
Add New Port Riptale

### PR DESCRIPTION
**Pull request by MadShmupper**
# New Port for { Riptale }

## Game Information
- **Title**: { Riptale }
- **URL**: { https://store.steampowered.com/app/616000/Riptale/ }

### CFW Tests
Ensure your game has been tested on all major CFWs:
- [x] AmberELEC
- [x] ArkOS
- [x] ROCKNIX
- [x] muOS

### Resolution Tests
Test all major resolutions:
- [x] 480x320
- [x] 640x480
- [x] 720x720 (RGB30)
- [x] Higher resolutions (e.g., 1280x720)

## File Structure
- Your port should have the following structure:
  - portname/
    - port.json
    - README.md
    - screenshot.jpg
    - cover.jpg
    - gameinfo.xml
    - Port Name.sh
    - portname/
      - <portfiles here>

## Additional Resources
For an in-depth guide on creating a pull request, refer to: [PortMaster Game Packaging Guide](https://portmaster.games/packaging.html#creating-a-pull-request)

### CODE REVIEW

**Review by kloptops — APPROVED** [review]
Looks good. 👍

File: ports/riptale/Riptale.sh [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+export PORT_32BIT="Y"
+
+get_controls
+[ -f "${controlfolder}/mod_${CFW_NAME}.txt" ] && source "${controlfolder}/mod_${CFW_NAME}.txt"
+
+$ESUDO chmod 666 /dev/tty0
+
+GAMEDIR="/$directory/ports/riptale"
+
+export LD_LIBRARY_PATH="/usr/lib32:$GAMEDIR/libs"
```
Comment (Cebion) [inline ports/riptale/Riptale.sh:26]: LD_LIBRARY path is overwriting system library exports.

This part needs to go to the end
```:$LD_LIBRARY_PATH"```

So 

```
export LD_LIBRARY_PATH="/usr/lib32:$GAMEDIR/libs:$LD_LIBRARY_PATH"
```

File: ports/riptale/port.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  "items_opt": [],
+  "attr": {
+    "title": "Riptale",
+    "porter": [
+      "Slayer366",
+      "MadShmupper"
+    ],
+    "desc": "Riptale is a challenging 2D roguelike action game aiming to make sword combat be as bloody and impactful as possible. The player collects gems that allow them to execute different attacks in the order of their choosing. All this results in a symphony of combos that fill the screen with blood and guts.",
+    "desc_md": null,
+    "inst": "Purchase from Steam https://store.steampowered.com/app/616000/Riptale/ Using only the latest Steam or Windows version, locate game.unx/data.win, audiogroup1.dat & audiogroup2.dat, plus all 8 folders- cavern_levels, challenge_rooms, crypt_levels, establishments, fonts, forest_levels, other, & ruin_levels. Place each of those files and folders inside the gamedata folder.\n   (Note: no need to rename game.unx)\n",
```
Comment (Cebion) [inline ports/riptale/port.json:17]: The inst field is only shown in the PortMaster app, has limited view for instructions. So putting full links in there might not b the best idea.

The inst_md field on the other hand is for the Instructions on the game details page where you can explain as long as you like. 

So on app make it short and snappy and on the website the full blown thing.

**Review by Cebion — CHANGES_REQUESTED** [review]

File: ports/riptale/Riptale.sh (hunk 1 above)
Comment (Slayer366) [inline ports/riptale/Riptale.sh:26]:
```suggestion
export LD_LIBRARY_PATH="/usr/lib32:$GAMEDIR/libs:$LD_LIBRARY_PATH"
```
Exactly as Cebion suggested, this is what we should have in the event that LD_LIBRARY_PATH already has a previously set value ^.  Admittedly, I should've caught this.

**Review by Slayer366 — CHANGES_REQUESTED** [review]

**Review by Slayer366 — APPROVED** [review]

**Review by Cebion — APPROVED** [review]

**Pull request merged**